tests/Unit/Http/Form/SignupFormTest.php

```diff
@@ -286,7 +286,58 @@ public function lastNameProvider(): array
     }
 
     /**
-     * Test that the Joind.in URL is being validated correctly
+     * Test that the joind.in username is being validated correctly
+     *
+     * @test
+     *
+     * @param string $joindin_username
+     * @param bool   $expectedResponse
+     * @dataProvider joindInUsernameProvider
+     */
+    public function joindInUsernameIsValidatedCorrectly($joindin_username, $expectedResponse)
+    {
+        $data['joindin_username'] = $joindin_username;
+
+        $form = new \OpenCFP\Http\Form\SignupForm($data, $this->purifier);
+        $form->sanitize();
+
+        $this->assertSame(
+            $expectedResponse,
+            $form->validateJoindInUsername(),
+            'Did not validate joind.in username as expected'
+        );
+    }
+
+    /**
+     * Data provider for joindInUsernameIsValidatedCorrectly
+     *
+     * @return array
+     */
+    public function joindInUsernameProvider(): array
+    {
+        $longUsername = '';
+
+        for ($x = 1; $x <= 256; ++$x) {
+            $longUsername .= 'X';
```

Comment:
I think this can easily be replace with

```php
$longUserName = str_pad('', 255, 'X');
```

or

```
$longUsername = str_repeat('X', 255);
```

Ha, sorry, I just realized it's moved from existing code. Do you mind adjusting, though?

Maybe it would even be more interesting to test the edges?

* length `100` is ok
* length `101` is not ok

What do you think?

Reply:
Yeah, I can make adjustments here tonight!